apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx

```diff
@@ -260,14 +260,15 @@ export const StepBasedFlow: FunctionComponent<AuthenticationFlowPropsInterface>
      */
     useEffect(() => {
 
-        const noOptionsSelected: boolean = authenticationSteps.length === 1
-            && authenticationSteps[ 0 ].options?.length === 0;
+        const steps: AuthenticationStepInterface[] = [ ...authenticationSteps ];
+
+        const optionsSelected = steps.find((step) => !isEmpty(step.options));
```

Comment:
Shall we use the `some` function instead of `find`. Since we just need a boolean value.
Also rename the variable to a meaningful one and add a doc comment to the `useEffect` explaining what is being done here.

```suggestion
        const hasStepsWithNoOptions: boolean = steps.some((step: AuthenticationStepInterface) => !isEmpty(step.options));
```

Reply:
fixed with 052d27d082a023d8b016ca28fcbf5e67a7709684